src/source/Ice/ConnectionListener.c

```diff
@@ -116,17 +126,28 @@ STATUS connectionListenerAddConnection(PConnectionListener pConnectionListener,
 STATUS connectionListenerRemoveConnection(PConnectionListener pConnectionListener, PSocketConnection pSocketConnection)
 {
     STATUS retStatus = STATUS_SUCCESS;
+    BOOL locked = FALSE, socketConnectionActive = FALSE;
 
     CHK(pConnectionListener != NULL && pSocketConnection != NULL, STATUS_NULL_ARG);
     CHK(!ATOMIC_LOAD_BOOL(&pConnectionListener->terminate), retStatus);
 
+    MUTEX_LOCK(pConnectionListener->lock);
+    locked = TRUE;
+
+    CHK_STATUS(hashTableContains(pConnectionListener->activeSocketConnections, ((UINT64) pSocketConnection), &socketConnectionActive));
```

Comment:
I'm not sure if I missed something, but why can't we just check if the connection is NULL? Since we already nullified it in `freeSocketConnection`, https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/626aae97e93cf05d34b62a4ca44f22435d836ac1/src/source/Ice/SocketConnection.c#L91.

Reply:
yh. Please take a look at this PR: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/441. I will cancel this one.